### PR TITLE
S2 T1 - HTML

### PR DESCRIPTION
**Pull request by jesshamby**
added html form and inputs for adding a new item to the collection

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Creates a connection to the collectorapp database
+ *
+ * @return PDO
+ */
+    function createDBConnection(): PDO {
```
Comment (Roomsycakes01) [inline functions.php:7]: not sure if you were told to put this in a function but would this just be better to run out of the function?

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+/**
+ * Creates recipe cards which have a name as well as stats on cuisine, time and a link to the recipe
+ *
+ * @param array $recipes
+ * @return string
+ */
+    function createRecipeCards(array $recipes): string {
+        if (count($recipes)>0) {
```
Comment (Roomsycakes01) [inline functions.php:33]: if you're checking for an array use is_array()

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            $recipeCards = '';
+            foreach ($recipes as $recipe) {
+                $recipeCards .= "<section class= 'recipe_card'>";
+                $recipeCards .= "<h2>{$recipe['recipe']}</h2>";
+                $recipeCards .= "<h3>Cuisine: {$recipe['cuisine']}</h3>";
+                $recipeCards .= "<h3>Time (mins): {$recipe['time']}</h3>";
+                $recipeCards .= "<a href= {$recipe['link']}>Link to recipe</a>";
+                $recipeCards .= "</section>";
```
Comment (Roomsycakes01) [inline functions.php:41]: can use single quotes here

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            }
+            return $recipeCards;
+        } else {
+            return 'No available recipes';
+        }
+    }
```
Comment (Roomsycakes01) [inline functions.php:91]: extra line

File: tests/functionsTest.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+require '../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class functionsTest extends TestCase
+{
+    function testSuccessCreateRecipeCards()
+    {
+$expected = "<section class= 'recipe_card'><h2>eggs</h2><h3>Cuisine: all</h3><h3>Time (mins): 1</h3><a href= https://www.google.com/>Link to recipe</a></section>";
```
Comment (Roomsycakes01) [inline tests/functionsTest.php:11]: weird indentation

File: main_page.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<?php
```
Comment (Roomsycakes01) [inline main_page.php:1]: Should this come before the doctype?

File: functions.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        return $recipes;
+    }
+
+/**
+ * Creates recipe cards which have a name as well as stats on cuisine, time and a link to the recipe
+ *
+ * @param array $recipes
+ * @return string
+ */
+    function createRecipeCards(array $recipes): string {
```
Comment (Roomsycakes01) [inline functions.php:32]: Caps on type hints i think

File: functions.php [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            foreach ($recipes as $recipe) {
+                $recipeCards .= "<section class= 'recipe_card'>";
+                $recipeCards .= "<h2>{$recipe['recipe']}</h2>";
+                $recipeCards .= "<h3>Cuisine: {$recipe['cuisine']}</h3>";
+                $recipeCards .= "<h3>Time (mins): {$recipe['time']}</h3>";
+                $recipeCards .= "<a href= {$recipe['link']}>Link to recipe</a>";
+                $recipeCards .= "</section>";
+            }
+            return $recipeCards;
+        } else {
```
Comment (Roomsycakes01) [inline functions.php:44]: extra line here

**Review by Roomsycakes01 — COMMENTED** [review]
It looks good, really had to nitpick this to find anything to say. Good work! :)

File: main_page.php [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    <meta charset="UTF-8">
+    <title>My Recipes</title>
+</head>
+<body>
+    <?php
+        echo createRecipeCards($recipes);
+    ?>
+    <form method="post">
+        <label for="recipe">Recipe:</label>
+        <input name="recipe" type="text">
```
Comment (EleanorRoberts) [inline main_page.php:36]: Consider putting inputs into label tags as best practice (as told by Ash)

File: tests/functionsTest.php [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        $inputa = [['recipe' => 'eggs', 'cuisine' => 'all', 'time' => '1', 'link' => 'https://www.google.com/']];
+
+        $case = createRecipeCards($inputa);
+
+        $this->assertEquals($expected, $case);
+    }
+
+    function testSuccessCreateRecipeCards2()
+    {
+        $expected = "No available recipes";
```
Comment (EleanorRoberts) [inline tests/functionsTest.php:22]: Single quotes here please 💃

File: tests/functionsTest.php [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        $inputa = [];
+
+        $case = createRecipeCards($inputa);
+
+        $this->assertEquals($expected, $case);
+    }
+
+    function testMalformedCreateRecipeCards()
+    {
+        $inputa = "hello";
```
Comment (EleanorRoberts) [inline tests/functionsTest.php:33]: Single quotes here as well :)

**Review by EleanorRoberts — COMMENTED** [review]
Looks great Jess! It's making me hungry thinking of all this yummy food 😂 
Just nitpicking at this point :)

File: functions.php (hunk 1 above)
Comment (jesshamby) [reply functions.php:7]: Was told to create it so I can use it again later :)

File: functions.php (hunk 2 above)
Comment (jesshamby) [reply functions.php:33]: It is an array- I'm checking to see if it's empty or not :)

File: tests/functionsTest.php (hunk 5 above)
Comment (jesshamby) [reply tests/functionsTest.php:11]: It was done by phpStorm because the line is so long

File: main_page.php (hunk 6 above)
Comment (jesshamby) [reply main_page.php:1]: Think so - this page has been trainer reviewed :)

File: functions.php (hunk 7 above)
Comment (jesshamby) [reply functions.php:32]: Think I checked PS-12 and it doesn't say about caps

**Pull request merged**